cmd/kibosh/main.go

```diff
@@ -74,7 +76,16 @@ func main() {
 	helmClientFactory := helm.NewHelmClientFactory(brokerLogger)
 	serviceAccountInstallerFactory := k8s.NewServiceAccountInstallerFactory(brokerLogger)
 
-	serviceBroker := broker.NewPksServiceBroker(conf, clusterFactory, helmClientFactory, serviceAccountInstallerFactory, charts, brokerLogger)
+	mapServiceInstanceToCluster := state.NewKeyValueStore()
+	err = mapServiceInstanceToCluster.Open(conf.StateDir)
+
+	if err != nil {
+		brokerLogger.Fatal("Unable to open state db", err)
+	}
+
+	defer mapServiceInstanceToCluster.Close()
```

Comment:
The way we're shutting down (line 104 blocks forever), this defer is never going to be reached. Is this a problem for this library if the process doesn't explicitly close, and instead terminates?

Reply:
Does it really block forever, or does it properly handle a sigterm or some other signal cause it to exit?

Either way, badger seems to have taken crash handling seriously - https://blog.dgraph.io/post/alice/

Reply:
A quick test and some google-fu seem to indicate that ListenAndServe() does not gracefully handle sigterm or sigint.  Boo!